core/src/main/java/com/microsoft/applicationinsights/internal/shutdown/SDKShutdownActivity.java

```diff
@@ -77,6 +77,20 @@ public void stop(long timeout, TimeUnit timeUnit) {
             });
         }
 
+        private volatile long perThreadTimeout = 10L;
+        private volatile TimeUnit perThreadTimeUnit = TimeUnit.SECONDS;
+
+        public synchronized long getPerThreadTimeout() {
+            return perThreadTimeout;
+        }
+        public synchronized TimeUnit getPerThreadTimeUnit() {
+            return perThreadTimeUnit;
+        }
+        public synchronized void setPerThreadTimeout(long timeout, TimeUnit unit) {
```

Comment:
same here I think we don't need synchronized on this methods as we are not having it configurable atm. Later when we think there is a need to have these configurable we can bring in synchronized. I don't think it makes sense to add overheads for the future :)